pymeasure/instruments/rohdeschwarz/sfm.py

```diff
@@ -0,0 +1,1444 @@
+        )
+
+    serial_bits = Instrument.control(
+        "SYST:COMM:SER:BITS?",
+        "SYST:COMM:SER:BITS %g",
+        """ A int property that controls the number of bits used in serial communication
+
+        Possible values are: 7 or 8
+        """,
+        validator = strict_discrete_set,
+        values=[7,8],
+        )
+
+    serial_flowcontrol = Instrument.control(
+        "SYST:COMM:SER:PACE?",
+        "SYST:COMM:SER:PACE %s",
+        """ A string property that controls the serial handshake type used in serial communication
+
+        Possible values are:
+
+        ======  =======
+        Value   Meaning
+        ======  =======
+        NONE    no flow-control/handshake
+        XON     XON/XOFF flow-control
+        ACK     hardware handshake with RTS&CTS
+        ======  =======
+
+        """,
+        validator = strict_discrete_set,
+        values=["NONE","XON","ACK"],
+        )
+
+    serial_parity = Instrument.control(
+        "SYST:COMM:SER:PAR?",
+        "SYST:COMM:SER:PAR %s",
+        """ A string property that controls the parity type used for serial communication
+
+        Possible values are:
+
+        ======  =======
+        Value   Meaning
+        ======  =======
+        NONE    no parity
+        EVEN    even parity
+        ODD     odd parity
+        ONE     parity bit fixed to 1
+        ZERO    parity bit fixed to 0
+        ======  =======
+
+        """,
+        validator = strict_discrete_set,
+        values=["NONE","EVEN","ODD","ONE","ZERO"],
+        )
+
+    serial_stopbits = Instrument.control(
+        "SYST:COMM:SER:SBIT?",
+        "SYST:COMM:SER:SBIT %g",
+        """ A int property that controls the number of stop-bits used in serial communication,
+
+        Possible values are: 1 or 2
+        """,
+        validator = strict_discrete_set,
+        values=[1,2],
+        )
+
+    date = Instrument.measurement(
+        "SYST:DATE?",
+        """
+        A tuple property for the date of the RTC in the unit
+
+        """,
+        )
+
+
+    time = Instrument.measurement(
+        "SYST:TIME?",
+        """
+        A tuple property for the time of the RTC in the unit
+
+        """,
+        )
+
+
+    #Unit subsystem (3.6.9)
+    scale_volt = Instrument.control(
+        "UNIT:VOLT?",
+        "UNIT:VOLT %s",
+        """ A string property that controls the unit to be used for voltage entries on the unit
+
+        Possible values are:
+        AV,FV, PV, NV, UV, MV, V, KV, MAV, GV, TV, PEV, EV,
+        DBAV, DBFV, DBPV, DBNV, DBUV, DBMV, DBV, DBKV, DBMAv, DBGV, DBTV, DBPEv, DBEV
+        """,
+        validator = strict_discrete_set,
+        values= ["AV","FV", "PV", "NV", "UV", "MV", "V", "KV", "MAV", "GV",
+                 "TV", "PEV", "EV", "DBAV", "DBFV", "DBPV", "DBNV", "DBUV",
+                 "DBMV", "DBV", "DBKV", "DBMAv", "DBGV", "DBTV", "DBPEv", "DBEV"],
+        )
+
+    def check_errors(self):
+        """ Read all errors from the instrument."""
```

Comment:
I think this method is not needed since it's already defined in the parent class

Reply:
Yes thats correct, as this is a SCPI complient unit,.
will remove it.

Reply:
Actually looking at isntryument.py and some other files, I could not find a definition for check_errors().
Only id, status and complete are predefined if SCPI is true.

Reply:
It's [here](https://github.com/pymeasure/pymeasure/blob/0880d5bd27b291510210251fc20acf0517bebaa7/pymeasure/instruments/instrument.py#L331) (not yet released, only on master)

Reply:
@bilderbuchi I see, now this comment makes sense.
Thanks for the pointer in the right direction.

So I will remove the check_errors() as proposed with a reference to #453.